yarn-project/p2p/src/services/libp2p/libp2p_service.ts

```diff
@@ -494,6 +500,28 @@ export class LibP2PService<T extends P2PClientType = P2PClientType.Full> extends
     return result.recipients.length;
   }
 
+  protected preValidateReceivedMessage(msg: Message, msgId: string, source: PeerId) {
+    const getValidator = () => {
+      if (msg.topic === this.topicStrings[TopicType.tx]) {
+        return this.msgIdSeenValidators[TopicType.tx];
+      }
+      if (msg.topic === this.topicStrings[TopicType.block_attestation]) {
+        return this.msgIdSeenValidators[TopicType.block_attestation];
+      }
+      if (msg.topic === this.topicStrings[TopicType.block_proposal]) {
+        return this.msgIdSeenValidators[TopicType.block_proposal];
+      }
+    };
```

Comment:
Given we ignore messages that don't have a validator, I think we should throw here:

```suggestion
      }
      throw new Error('No message validator for topic: ' + msg.topic)
    };
```